src/Grpc.Net.Client/Internal/StreamExtensions.cs

```diff
@@ -107,16 +107,34 @@ private static async Task<(int length, bool compressed)?> ReadHeaderAsync(Stream
                 // If the message is larger then the array will be replaced when the message size is known.
                 buffer = ArrayPool<byte>.Shared.Rent(minimumLength: 4096);
 
-                var headerDetails = await ReadHeaderAsync(responseStream, buffer, cancellationToken).ConfigureAwait(false);
+                int read;
```

Comment:
Looks like the ReadHeaderAsync method is no longer used anywhere else after you relocated the logic. I think we can remove it.